src/AzureClient/Magic/ConnectMagic.cs

```diff
@@ -0,0 +1,107 @@
+                                Connect to an Azure Quantum workspace:
+                                ```
+                                In []: %connect {ParamName_SubscriptionId}=SUBSCRIPTION_ID
+                                                {ParamName_ResourceGroupName}=RESOURCE_GROUP_NAME
+                                                {ParamName_WorkspaceName}=WORKSPACE_NAME
+                                                {ParamName_StorageAccountConnectionString}=CONNECTION_STRING
+                                Out[]: Connected to WORKSPACE_NAME
+                                ```
+                            ".Dedent(),
+
+                            $@"
+                                Connect to an Azure Quantum workspace and force a credential prompt:
+                                ```
+                                In []: %connect {ParamName_Login}
+                                                {ParamName_SubscriptionId}=SUBSCRIPTION_ID
+                                                {ParamName_ResourceGroupName}=RESOURCE_GROUP_NAME
+                                                {ParamName_WorkspaceName}=WORKSPACE_NAME
+                                                {ParamName_StorageAccountConnectionString}=CONNECTION_STRING
+                                Out[]: To sign in, use a web browser to open the page https://microsoft.com/devicelogin
+                                        and enter the code [login code] to authenticate.
+                                        Connected to WORKSPACE_NAME
+                                ```
+                                Use the `{ParamName_Login}` option if you want to bypass any saved or cached
+                                credentials when connecting to Azure.
+                            ".Dedent()
+                        }
+                }) {}
+
+        /// <summary>
+        ///     Connects to an Azure workspace given a subscription ID, resource group name,
+        ///     workspace name, and connection string as a JSON-encoded object.
+        /// </summary>
+        public override async Task<ExecutionResult> RunAsync(string input, IChannel channel)
+        {
+            var inputParameters = ParseInputParameters(input);
+
+            var storageAccountConnectionString = inputParameters.DecodeParameter<string>(ParamName_StorageAccountConnectionString);
+            if (string.IsNullOrEmpty(storageAccountConnectionString))
+            {
+                return await AzureClient.PrintConnectionStatusAsync(channel);
+            }
+
+            var subscriptionId = inputParameters.DecodeParameter<string>(ParamName_SubscriptionId);
+            var resourceGroupName = inputParameters.DecodeParameter<string>(ParamName_ResourceGroupName);
+            var workspaceName = inputParameters.DecodeParameter<string>(ParamName_WorkspaceName);
+            var forceLogin = inputParameters.DecodeParameter<bool>(ParamName_Login);
```

Comment:
That's an optional parameter, right?
What happens in this line when it's not found - should we assign a default (false)?

Reply:
Yes, `DecodeParameter` implicitly assigns a default value of false here if it's not found, but as Chris also suggested, I'll add an explicit parameter here and explicitly specify that the default value of this one is false.

---
In reply to: [426997830](https://github.com/microsoft/iqsharp/pull/144#discussion_r426997830) [](ancestors = 426997830)